src/west/_bootstrap/main.py

```diff
@@ -247,22 +250,41 @@ def append_to_pythonpath(directory):
 
 def wrap(argv):
     printing_version = False
-
-    if argv and argv[0] in ('-V', '--version'):
-        print('West bootstrapper version: v{} ({})'.format(version.__version__,
-                                                    os.path.dirname(__file__)))
-        printing_version = True
+    printing_help_only = False
+
+    if argv:
+        if argv[0] in ('-V', '--version'):
+            print('West bootstrapper version: v{} ({})'.
+                  format(version.__version__, os.path.dirname(__file__)))
+            printing_version = True
+        elif len(argv) == 1 and argv[0] in ('-h', '--help'):
+            # This only matters if we're called outside of an
+            # installation directory. We delegate to the main help if
+            # called from within one, because it includes a list of
+            # available commands, etc.
+            printing_help_only = True
 
     start = os.getcwd()
     try:
         topdir = find_west_topdir(start)
     except WestNotFound:
         if printing_version:
             sys.exit(0)         # run outside of an installation directory
+        elif printing_help_only:
+            # We call print multiple times here and below instead of using
+            # \n to be newline agnostic.
+            print('To set up a Zephyr installation here, run "west init".')
+            print('Run "west init -h" for additional information.')
+            sys.exit(0)
         else:
-            sys.exit('Error: not a Zephyr directory (or any parent): {}\n'
-                     'Use "west init" to install Zephyr here'
-                     .format(os.getcwd()))
+            print('Error: "{}" is not a Zephyr installation directory.'.
+                  format(start), file=sys.stderr)
+            print('Things to try:', file=sys.stderr)
+            print(' - Run "west init" to set up an installation here.',
+                  file=sys.stderr)
+            print(' - Run "west init -h" for additional information.',
```

Comment:
Much better.

One comment here though.
There are not many commands the user can actually do, currently only `west init`, so instead of informing the user that additional help is available, when writing `west init -h`, we could potentially just concatenate `west init -h` to him, as that is most like is what he would need, as he already wrote: `west -h`

Maybe:
```
$ west -h
To set up a Zephyr installation here, run "west init".

usage: west init [-h] [-b BASE_URL] [-m MANIFEST_URL] [--mr MANIFEST_REV]
                 [-w WEST_URL] [--wr WEST_REV]
                 [directory]

Bootstrap initialize a Zephyr installation

positional arguments:
  directory             Initializes in this directory, creating it if
                        necessary

optional arguments:
  -h, --help            show this help message and exit
  -b BASE_URL, --base-url BASE_URL
                        Base URL for both 'manifest' and 'zephyr'
                        repositories; cannot be given if either -m or -w are
  -m MANIFEST_URL, --manifest-url MANIFEST_URL
                        Zephyr manifest fetch URL, default
                        https://github.com/zephyrproject-rtos/manifest
  --mr MANIFEST_REV, --manifest-rev MANIFEST_REV
                        Manifest revision to fetch, default master
  -w WEST_URL, --west-url WEST_URL
                        West fetch URL, default
                        https://github.com/zephyrproject-rtos/west
  --wr WEST_REV, --west-rev WEST_REV
                        West revision to fetch, default master
```

Reply:
+1

Reply:
> Otherwise it breaks the convention that, in order to get help for a particular option, you invoke it with west <option> -h.

In principle I agree with you on this. But I decided to make the comment anyway, as `west init -h` is fact the only help message available here, and thus could be nice to present it.

So i'm fine with both options, but wanted to raise it in order to get others opinion.

Reply:
I thought about that, but I figured it would be confusing to users who -- reasonably -- assume that `pip install west` installs all of west. I know we have been careful to document that isn't true, but most people won't read that.

So I'm trying to balance helpfulness with not suggesting there's only one option here.